detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/VarCouldBeValSpec.kt

```diff
@@ -50,6 +51,39 @@ class VarCouldBeValSpec(val env: KotlinCoreEnvironment) {
 
             assertThat(subject.compileAndLintWithContext(env, code)).isEmpty()
         }
+
+        @Test
+        fun `does not report private variables that are re-assigned inside an unknown type`() {
+            val code = """
+            private var a = 1
+
+            fun foo() {
+                with(UnknownReference) {
+                    a = 2
+                }
+            }
+            """
+
+            assertThat(subject.lintWithContext(env, code)).isEmpty()
+        }
+
+        @Test
+        fun `reports private variables that have the same name as those re-assigned within a known type`() {
+            val code = """
+            class MyClass(var a: Int)
+
+            private var a = 1
+            private val myObj = MyClass(1)
+
+            fun foo() {
+                with(myObj) {
+                    a = 2
+                }
+            }
+            """
+
+            assertThat(subject.lintWithContext(env, code)).hasSize(1)
```

Comment:
This case should compile

```suggestion
            assertThat(subject.compileAndLintWithContext(env, code)).hasSize(1)
```